src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2401,6 +2393,16 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
                                 break;
                             }
 
+                            //EIP-3860
+                            if (spec.IsEip3860Enabled && initCodeLength > spec.MaxInitCodeSize)
+                            {
+                                //currently this needs to update nonce - may be a change in spec
+                                _state.IncrementNonce(env.ExecutingAccount);
```

Comment:
hmmm I think that we're increasing the nonce at the end of every transaction, so I am bit suprised that we had to add it here. Let's discuss it